bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/FontData.java

```diff
@@ -240,6 +240,52 @@ public FontData(String string) {
 	}
 }
 
+
+/**
+ * Constructs a new FontData copy
+ *
+ * @param fontData the FondData object for which copy is needed to be made
+ *
+ * @exception IllegalArgumentException
+ * <ul>
+ *    <li>ERROR_NULL_ARGUMENT - if the argument is null</li>
+ * </ul>
+ * @since 3.130
+ */
+public FontData(FontData fontData) {
+	if (fontData == null)
+		SWT.error(SWT.ERROR_NULL_ARGUMENT);
+
+	if (fontData.data != null) {
+		LOGFONT newData = new LOGFONT();
+		setHeight(fontData.getHeightF());
```

Comment:
Instead of setting specific fields via setters, shouldn't we copy over all data in a proper copy constructor, i.e., also the height, lang, country and variant fields? This seems to be a copy-over from the String-based constructor, in which some original data is missing and needs to be re-initialized from the data being given using according setters.